torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -55,6 +55,16 @@
 static_input_logger = getArtifactLogger("torch._dynamo", "cudagraph_static_inputs")
 
 
+# Workaround of https://github.com/pytorch/pytorch/issues/62027
+# tensor.contiguous() guarantees to return non-zero sorted-stride,
+# tensor.to(memory_format=torch.contiguous_format) can keep zero strides.
```

Comment:
hmm... can you talk more about why you think it's worth trying to handle the zero strides case? 

The common case where I imagine zero strides showing up is when you compile a model (not including the loss), and then run `.sum().backward()`. The autograd engin will do something like `torch.ones(1).expand(tangent_shape)`, and the incoming tangent input to the backward will have a zero stride.

We are still kind of out-of-luck in this case though, because the strides of the tangent will be **different** than the strides of the forward graph output (the tangent technically has overlapping memory, while the forward output does not).

We will have been forced to trace out a backward graph ahead-of-time that assumed that the tangent was a plain contiguous tensor, so it does kind of feel like we are forced to emit the contiguous call in the backward.

Is the zero-stride-handling here attempting to handle that case? Or something else

Reply:
Copy from chat:

Inductor generates asserts for sizes and strides, and as output strides were non-zero - then it generates non-zero assert.
When at runtime tangents come with zero stride and .to(contiguous) does not change them - it fails on inductor assert.